contracts/Unibright.sol

```diff
@@ -2,288 +2,251 @@
-    event WhitelistTokenUpdated(
-        address triggerredBy,
-        address token,
-        bool isWhitelisted
-    );
 
-    event ERC20PaymentReleased(
+    event UbtPaymentReleased(
         IERC20 indexed token,
         address revenueAddress,
         address stakingAddress,
         uint256 amount
     );
 
-    event ERC20Deposit(
+    event UbtDeposited(
         address sender,
         address token,
         uint256 tokenAmount,
-        uint256 state_lastEventNonce,
+        uint256 lastEventNonce,
         string destinationAddress
     );
 
     uint256 public totalShares;
-    uint256 public state_lastEventNonce;
+    uint256 public lastEventNonce;
 
     mapping(address => uint256) public shares;
-    mapping(address => address) public validatorStakingAddress; // revenueAddress => validatorStakingAddress
-    address[] public payees;
+    mapping(address => address) public stakingAddresses;
+    mapping(address => uint256) public ubtReleased;
 
-    mapping(IERC20 => uint256) public erc20TotalReleased;
-    mapping(IERC20 => mapping(address => uint256)) public erc20Released;
-    
-    address public theWhitelistedToken;
+    mapping(address => bool) public payees;
 
-    uint256 public theErc20ToBeReleasedInPeriod;
-    uint256 public theErc20NotReleasedInLastPeriod;
-    uint256 public theErc20CurrentPeriod;
-
-    mapping(uint256 => mapping (address => uint256)) theErc20ReleasedPerRecipientInPeriods;
+    uint256 public ubtTotalReleased;
+    mapping(uint256 => mapping (address => uint256)) public ubtReleasedPerRecipientInPeriods;
+    
+    uint256 public ubtToBeReleasedInPeriod;
+    uint256 public ubtNotReleasedInLastPeriods;
+    uint256 public ubtCurrentPeriod;
 
+    address public whitelistedToken;
 
-    /**
-     * @dev Creates an instance of `UBTSplitter` where each account in `payees` is assigned the number of shares at
-     * the matching position in the `shares` array.
-     *
-     * All addresses in `payees` must be non-zero. Both arrays must have the same non-zero length, and there must be no
-     * duplicates in `payees`.
-     */
     constructor(address token) {
-        theWhitelistedToken = token;
+        whitelistedToken = token;
     }
 
     /**
      * @dev Modifier for checking for zero address
      */
-    modifier zeroAddress(address token) {
-        require(token != address(0), "UBTSplitter: Address is zero address");
+    modifier zeroAddress(address address_) {
+        require(address_ != address(0), "address is zero address");
         _;
     }
 
     /**
      * @dev Modifier for checking for empty string
      */
-    modifier emptyString(string memory baseledgervaloper) {
-        bytes memory tempEmptyStringTest = bytes(baseledgervaloper);
+    modifier emptyString(string memory str) {
+        bytes memory tempEmptyStringTest = bytes(str);
         require(
             tempEmptyStringTest.length != 0,
-            "UBTSplitter: String is empty"
+            "string is empty"
```

Comment:
do we need (it is good practice) to check if address has proper prefix?
eg. baseledgerDestinationAddress should start with "baseledger..."
baseledgerValidatorAddress should start with "baseledgervaloper..."
this will fail in baseledger anyways, just to double check if we should add it here as well

Reply:
I agree that we should add this check, there are different addresses format in cosmos so we can expect some confusion from the transaction initiators

Reply:
should be a separate task imo to be able to merge all current state to develop, not sure how startsWith is working in solidity and how efficient is, would need to take a look